libtextsecure/test/protocol_test.js

```diff
@@ -40,193 +40,5 @@ describe('Protocol', function() {
-                            getKeysForNumberMap["SNOWDEN"] = data.getKeys;
-
-                        var checkMessage = function() {
-                            var msg = messagesSentMap["SNOWDEN.1"];
-                            delete messagesSentMap["SNOWDEN.1"];
-                            //XXX: This should be all we do: isEqual(data.expectedCiphertext, msg.body, false);
-                            if (msg.type == 1) {
-                                var expected = getString(data.expectedCiphertext);
-                                var decoded = textsecure.protobuf.WhisperMessage.decode(expected.substring(1, expected.length - 8), 'binary');
-                                var result = getString(msg.body);
-                                return getString(decoded.encode()) == result.substring(1, result.length - 8);
-                            } else {
-                                var decoded = textsecure.protobuf.PreKeyWhisperMessage.decode(getString(data.expectedCiphertext).substr(1), 'binary');
-                                var result = getString(msg.body).substring(1);
-                                return getString(decoded.encode()) == result;
-                            }
-                        }
-
-                        if (data.endSession)
-                            return textsecure.messaging.closeSession("SNOWDEN").then(checkMessage);
-                        else
-                            return textsecure.messaging.sendMessageToNumber("SNOWDEN", data.smsText, [], Date.now()).then(checkMessage);
-                    }
-
-                    if (data.ourBaseKey !== undefined)
-                        privKeyQueue.push(data.ourBaseKey);
-                    if (data.ourEphemeralKey !== undefined)
-                        privKeyQueue.push(data.ourEphemeralKey);
-
-                    if (data.ourIdentityKey !== undefined)
-                        return textsecure.crypto.createKeyPair(data.ourIdentityKey).then(function(keyPair) {
-                            textsecure.storage.putEncrypted("25519KeyidentityKey", keyPair);
-                            return postLocalKeySetup();
-                        });
-                    else
-                        return postLocalKeySetup();
-
-                default:
-                    return Promise.resolve(false);
-                }
-            }
-            return doStep().then(stepDone);
-        };
-
-        describe("test vectors", function() {
-            for (var i in axolotlTestVectors) {
-                it(axolotlTestVectors[i].name, function(done) {
-                    localStorage.clear();
-                    return runAxolotlTest(axolotlTestVectors[i].vectors).then(function(res) {
-                        assert(res);
-                    }).then(done).catch(done);
-                });
-            }
-        });
-    });
+    // TODO: Use fake_api's hiding of api.sendMessage to test sendmessage.js' maze
```

Comment:
Either that or write a fake_server that can simply map api paths to configurable canned responses. I imagine npm or bower has a package like that somewhere.